osu.Game.Rulesets.Taiko/Difficulty/TaikoDifficultyCalculator.cs

```diff
@@ -46,13 +46,13 @@ public TaikoDifficultyCalculator(IRulesetInfo ruleset, IWorkingBeatmap beatmap)
 
         protected override IEnumerable<DifficultyHitObject> CreateDifficultyHitObjects(IBeatmap beatmap, double clockRate)
         {
-            List<TaikoDifficultyHitObject> taikoDifficultyHitObjects = new List<TaikoDifficultyHitObject>();
+            List<DifficultyHitObject> taikoDifficultyHitObjects = new List<DifficultyHitObject>();
 
             for (int i = 2; i < beatmap.HitObjects.Count; i++)
             {
                 taikoDifficultyHitObjects.Add(
                     new TaikoDifficultyHitObject(
-                        beatmap.HitObjects[i], beatmap.HitObjects[i - 1], beatmap.HitObjects[i - 2], clockRate, i
+                        beatmap.HitObjects[i], beatmap.HitObjects[i - 1], beatmap.HitObjects[i - 2], clockRate, taikoDifficultyHitObjects, i - 2
```

Comment:
I'll just note that the change in the last argument very luckily does not seem like it will influence results, because the only usage of the old `ObjectIndex` was to calculate a difference between two object indices in `Rhythm`, and so `b.ObjectIndex - a.ObjectIndex == (b.Position + 2) - (a.Position + 2)`.